### PR TITLE
Fixes issue where glance fails to parse image details with milli seconds in the date

### PR DESCRIPTION
**Pull request by arvindn05**
Fixes issue where glance fails to parse image details with milli seconds in the date.
The issue is described in this bug. https://issues.apache.org/jira/browse/JCLOUDS-333

Caused by: java.text.ParseException: Unparseable date: "2015-06-19T15:06:58.000+0000"
at java.text.DateFormat.parse(Unknown Source)
at org.jclouds.date.internal.SimpleDateFormatDateService.iso8601SecondsDateParse(SimpleDateFormatDateService.java:164)
... 17 more

### CODE REVIEW

File: openstack-glance/src/test/java/org/jclouds/openstack/glance/v1_0/functions/ParseImageDetailsFromHeadersTest.java [hunk 1]
```diff
@@ -32,51 +32,80 @@
...
+                .diskFormat(DiskFormat.RAW)
+                .checksum("233afa7b8809d840679b5f0d36d7350a")
+                .size(65645798l)
+                .status(Image.Status.ACTIVE)
+                .owner("5821675")
+                .isPublic(true)
+                .createdAt(new SimpleDateFormatDateService().iso8601DateOrSecondsDateParse("2012-05-18T18:06:44"))
+                .updatedAt(new SimpleDateFormatDateService().iso8601DateOrSecondsDateParse("2012-05-18T18:42:58"))
+                .build();
+    }
```
Comment (nacx) [inline openstack-glance/src/test/java/org/jclouds/openstack/glance/v1_0/functions/ParseImageDetailsFromHeadersTest.java]: jclouds uses a [3 space indentation](https://cwiki.apache.org/confluence/display/JCLOUDS/Coding+Standards). Could you revert the re-formatting of this class so we can see the real diff?

**Comment by zack-shoylev** [comment]
Outside of some formatting issues and the merge commit (is it needed?), this looks good. Good bug to fix. Thanks @arvindn05

**Comment by arvindn05** [comment]
Thanks Zack and nac. Sorry for the formatting issues, this is my first commit so please bear with me :)

The merge commit is not needed. Trying to learn working with github.

Please see my latest review. Thanks!

**Comment by arvindn05** [comment]
anything i need to do before this change can be merged @zack-shoylev ?

**Comment by zack-shoylev** [comment]
Rebase to a single commit

**Comment by arvindn05** [comment]
I am not sure if the rebase worked....if not i can just create another clean pull request without the many commits. please let me know @zack-shoylev

**Comment by zack-shoylev** [comment]
Hmm commits seem messed up. If you want to, you can make a new one. Just make sure to reference it here. Thanks!

**Comment by arvindn05** [comment]
created a new pull request with just the one commit. @zack-shoylev 
https://github.com/jclouds/jclouds-labs-openstack/pull/197

**Comment by zack-shoylev** [comment]
Thanks, I will merge and backport in a bit.

**Pull request closed without merging**